repo/packages/B/beta-tensorflow/1/config.json

```diff
@@ -0,0 +1,329 @@
+        "tf_image": {
+          "title": "TensorFlow Docker Image",
+          "description": "Name of the TensorFlow Docker Image in which Worker/PS tasks will be run (usr/img:version).",
+          "type": "string",
+          "default": "mesosphere/dcos-tensorflow:2.7.4-1.3.0"
+        },
+        "gpu_tf_image": {
+          "title": "TensorFlow GPU Docker Image",
+          "description": "Name of the TensorFlow Docker Image in which GPU Worker tasks will be run (usr/img:version).",
+          "type": "string",
+          "default": "mesosphere/dcos-tensorflow:2.7.4-1.3.0-gpu"
+        }
+      },
+      "required": [
+        "name",
+        "user",
+        "job_url",
+        "job_name",
+        "tf_image",
+        "gpu_tf_image"
+      ]
+    },
+    "parameter_server": {
+      "description": "TF Parameter Server configuration properties",
+      "type": "object",
+      "properties": {
+        "count": {
+          "title": "Node count",
+          "description": "Number of TF ps tasks to run",
+          "type": "integer",
+          "default": 2
+        },
+        "placement_constraint": {
+          "title": "Placement constraint",
+          "description": "Marathon-style placement constraint for nodes. Example: 'hostname:UNIQUE'",
+          "type": "string",
+          "default": ""
+        },
+        "port": {
+          "title": "Port",
+          "description": "The port Parameter Servers will use for gRPC.",
+          "type": "integer",
+          "default": 2223
+        },
+        "cpus": {
+          "title": "CPU count",
+          "description": "TF PS CPU requirements",
+          "type": "number",
+          "default": 1.0
+        },
+        "mem": {
+          "title": "Memory size (MB)",
+          "description": "TF PS mem requirements (in MB)",
+          "type": "integer",
+          "default": 4096
+        },
+        "disk": {
+          "title": "Disk size (MB)",
+          "description": "TF PS persistent disk requirements (in MB)",
+          "type": "integer",
+          "default": 4096
+        },
+        "disk_type": {
+          "title": "Disk type [ROOT, MOUNT]",
+          "description": "Mount volumes require preconfiguration in DC/OS",
+          "enum": [
+            "ROOT",
+            "MOUNT"
+          ],
+          "default": "ROOT"
+        }
+      },
+      "required": [
+        "count",
+        "cpus",
+        "mem",
+        "port",
+        "disk",
+        "disk_type"
+      ]
+    },
+    "worker": {
+      "description": "TF Worker configuration properties",
+      "type": "object",
+      "properties": {
+        "count": {
+          "title": "Node count",
+          "description": "Number of TF Workers to run",
+          "type": "integer",
+          "default": 3
+        },
+        "placement_constraint": {
+          "title": "Placement constraint",
+          "description": "Marathon-style placement constraint for nodes. Example: 'hostname:UNIQUE'",
```

Comment:
Just curious: Why do we have a placement constraint for gpu worker

Reply:
One needs to ensure that the GPU worker is started on a node with a GPU.

Reply:
Yes, but I might have further constraints or? e.g., hostname unique?

Reply:
Oh, wait. I see now. The GPU worker **DOESN'T** have a placement constraint.

Reply:
you could imagine limiting it to a specific GPU machine with one of these constraints.